DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -134,48 +130,162 @@ function ValidateProperties
+            {
+                Write-Verbose -Message ( @(
+                    "Default gateway does NOT match desired state. Expected $DefaultGateway, "
+                    "actual $($currentSettings['DefaultGateway'])."
+                    ) -join "" | Out-String
+                )
+                $requiresChanges = $true
+            }
+            else
+            {
+                Write-Verbose -Message "Default gateway is correct."
+            }
+        }
+
+        #Test if DHCP is already disabled
+        if(!(Get-NetIPInterface -InterfaceAlias $InterfaceAlias -AddressFamily `
+            $AddressFamily).Dhcp.ToString().Equals('Disabled'))
+        {
+            Write-Verbose -Message "DHCP is NOT disabled."
+            $requiresChanges = $true
+        }
+        else
+        {
+            Write-Verbose -Message "DHCP is already disabled."
+        }
+
+        if($requiresChanges)
+        {
+            # Apply is true in the case of set - target resource - in which case, it will apply the
+            # required IP configuration
             if($Apply)
             {
-                Write-Verbose -Message "Setting IPAddress ..."
+                Write-Verbose -Message ( @(
+                    "At least one setting differs from the passed parameters. Applying "
+                    "configuration..."
+                    ) -join "" | Out-String
+                )
+
+                # Build parameter hash table
+                $Parameters = @{}
                 $Parameters["IPAddress"] = $IPAddress
                 $Parameters["PrefixLength"] = $SubnetMask
                 $Parameters["InterfaceAlias"] = $currentIP[0].InterfaceAlias
+                if($DefaultGateway)
+                {
+                    $Parameters["DefaultGateway"] = $DefaultGateway
+                }
 
-                if($DefaultGateway){ $Parameters["DefaultGateWay"] = $DefaultGateway }
+                # If null, remove the gateway from the current settings hash table
+                if($currentSettings['DefaultGateway'] -eq $null)
+                {
+                    $currentSettings.remove('DefaultGateway')
+                }
+
+                # Remove IP address first; required if the IP is correct but other settings are not
+                Remove-NetIPAddress @currentSettings -confirm:$false -ErrorAction Stop
+
+                # Apply the specified IP configuration
                 $null = New-NetIPAddress @Parameters -ErrorAction Stop
 
                 # Make the connection profile private
-                Get-NetConnectionProfile -InterfaceAlias $InterfaceAlias | Set-NetConnectionProfile -NetworkCategory Private -ErrorAction SilentlyContinue
-                Write-Verbose -Message "IPAddress is set to $IPAddress."
+                Get-NetConnectionProfile -InterfaceAlias $InterfaceAlias | `
+                    Set-NetConnectionProfile -NetworkCategory Private -ErrorAction SilentlyContinue
```

Comment:
The Network Category should probably be specified as a parameter and validated seperately. As I can imagine not wanting to set it to Private.